OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java

```diff
@@ -442,10 +442,12 @@ private Location setCurrentLocation(Location currentLocation, boolean returnUpda
 				if (allowableDeviation == 0) {
 					allowableDeviation = getDefaultAllowedDeviation(settings, route.getAppMode(), posTolerance);
 				}
+				boolean isRecalculationActivated = allowableDeviation > 0 &&
```

Comment:
USE_ROUTE_RECALCULATION_DISTANCE it should be reverted, new setting is not needed same as changes to RoutingHelper it only possibly breaks existing functionality.